noodles-editor/src/noodles/fields.ts

```diff
@@ -1310,3 +1311,220 @@ export class BezierCurveField extends Field<z.ZodType<BezierCurveData>> {
     return segments
   }
 }
+
+// Helper to detect coordinate field names with various patterns (case-insensitive)
+// Recognizes: lat, latitude, lng, lon, longitude, and any prefixed versions like pickup_lat, dropoff_lng, etc.
+export function findCoordinateFields(obj: Record<string, unknown>): { lng: number; lat: number } | null {
+  if (typeof obj !== 'object' || obj === null) return null
+
+  // Patterns to match longitude and latitude field names (case-insensitive)
+  // Matches: lng, _lng, pickup_lng, dropoff_lon, start_longitude, etc.
+  const lngPatterns = /^(\w*_)?(lng|lon|longitude)$/i
+  const latPatterns = /^(\w*_)?(lat|latitude)$/i
+
+  let lng: number | undefined
+  let lat: number | undefined
+
+  for (const [key, value] of Object.entries(obj)) {
+    if (typeof value === 'number') {
+      if (!lng && lngPatterns.test(key)) {
+        lng = value
+      } else if (!lat && latPatterns.test(key)) {
+        lat = value
+      }
+    }
+  }
+
+  return lng !== undefined && lat !== undefined ? { lng, lat } : null
+}
+
+// Helper to convert a single point to GeoJSON Feature
+export function pointsToFeature(value: unknown): Feature | null {
+  if (!value) return null
+
+  // Already a GeoJSON Feature
+  if (typeof value === 'object' && 'type' in value && value.type === 'Feature') {
+    return value as Feature
+  }
+
+  // Single point as array: [lng, lat]
+  if (Array.isArray(value) && value.length === 2 && typeof value[0] === 'number') {
+    return turf.point([value[0], value[1]], value)
```

Comment:
The properties arg is typically an object
```
return turf.point([value[0], value[1]], { value })
```

Good idea to pass the original through.. lets just use a consistent key. Could be `value`, `source`, `input`